### PR TITLE
PHP 8.1 compatibility - interfaces Countable,ArrayAccess,Iterator,SeekableIterator,RecursiveFilterIterator,IteratorAggregate,Serializable

### PR DESCRIPTION
**Pull request by EugenePikalov**
PHP 8.1 compatibility
- added return type for Countable::count
- added return type for methods offsetExists,offsetGet,offsetSet,offsetUnset (interface ArrayAccess)
- added return type for methods rewind(),current(),key(),next(),valid() (interface Iterator)
- added return type for SeekableIterator::seek()
- added return type for methods getChildren(),hasChildren() (interface RecursiveFilterIterator)
- added return type for IteratorAggregate::getIterator()
- added return type for methods serialize(),unserialize() (interface Serializable)
- removed "return" in rewind(),next() methods

### CODE REVIEW

**Comment by gehrisandro** [comment]
@EugenePikalov very nice, thank you 👍

**Comment by quipo** [comment]
Zend_Controller_Action_HelperBroker_PriorityStack  and
Zend_View_Helper_Placeholder_Container_Standalone
need a similar fix too

**Comment by EugenePikalov** [comment]
> Zend_Controller_Action_HelperBroker_PriorityStack and Zend_View_Helper_Placeholder_Container_Standalone need a similar fix too

It's already in PR

**Comment by Shardj** [comment]
Sorry, just merged another PR which has created some conflicts

**Comment by Shardj** [comment]
I should also add, using mixed return type is no longer an option due to the demand to continue supporting PHP 7.4

See https://github.com/Shardj/zf1-future/issues/194 and the solution suggested by @hc-jworman with the `#[\ReturnTypeWillChange]` attribute

**Comment by Hikariii** [comment]
https://github.com/Shardj/zf1-future/pull/205 has the work done here AND maintains php 7.4 support.

File: library/Zend/Feed/Element.php [hunk 1]
```diff
@@ -409,16 +409,16 @@ public function offsetGet($offset)
      * @param  string $value
      * @return string
```
Comment (alexkuusk) [inline library/Zend/Feed/Element.php:410]: @return void ?

**Pull request closed without merging**